### PR TITLE
Add calculations of adjacent tiles with tests

### PR DESCRIPTION
**Pull request by patkkka**
Please check AdjacentTilesCalc.java and AdjacentTilesCalcTest.java
"Field" name was changed to "Tile" and generated some "false" new lines.

### CODE REVIEW

File: src/main/java/com/spanish_inquisition/battleship/common/AdjacentTilesCalc.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        LOWER_LEFT,
+        LOWER_RIGHT
+    }
+
+    public static List<Integer> getBoardIndexesOfAllAdjacentTilesOf(Integer tile) {
+        List<Integer> adjacentFields = new LinkedList<>();
+        TileDirection[] enumValues = TileDirection.values();
+        for (TileDirection tileDirection: enumValues) {
+            int adjacentBoardIndex = getAdjacentTileIndex(tile, tileDirection);
+            if(adjacentBoardIndex != -1) {
```
Comment (paulstarbi) [inline src/main/java/com/spanish_inquisition/battleship/common/AdjacentTilesCalc.java]: magic number :-1:

File: src/main/java/com/spanish_inquisition/battleship/common/AdjacentTilesCalc.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    public static int translateCoordinatesToIndex(int xCoordinate, int yCoordinate) {
+        if (xCoordinate < 0 || xCoordinate > IMPORTED_BOARD_SIZE - 1 || yCoordinate < 0 || yCoordinate > IMPORTED_BOARD_SIZE - 1) {
+            return -1;
+        }
+        return IMPORTED_BOARD_SIZE * (yCoordinate) + (xCoordinate);
+    }
+
+    public static int[] translateIndexToCoordinates(int index) {
+        int[] coordinates = new int[2];
+        coordinates[0] = (index % IMPORTED_BOARD_SIZE); //x coordinate
```
Comment (paulstarbi) [inline src/main/java/com/spanish_inquisition/battleship/common/AdjacentTilesCalc.java]: unnecessary braces

File: src/test/java/com/spanish_inquisition/battleship/common/AdjacentTilesCalcTest.java [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+                {10, new int[] {0,1}},
+                {11, new int[] {1,1}},
+                {38, new int[] {8,3}},
+                {65, new int[] {5,6}},
+                {99, new int[] {9,9}}
+        };
+    }
+
+    @DataProvider(name = "CoordinatesProvider")
+    public Object[][] coordinatesProvider(){
```
Comment (paulstarbi) [inline src/test/java/com/spanish_inquisition/battleship/common/AdjacentTilesCalcTest.java]: make private

File: src/main/java/com/spanish_inquisition/battleship/client/board/GameBoardBuilder.java [hunk 4]
```diff
@@ -61,14 +61,14 @@ private void addCorrespondingLabelsToTiles() {
...
-            int[] tileCoordinates = AdjacentFieldsCalc.translateIndexToCoordinates(boardIndex);
+            int[] tileCoordinates = AdjacentTilesCalc.translateIndexToCoordinates(boardIndex);
             tile.setLabels(horizontalLabels.get(tileCoordinates[1] + 1), verticalLabels.get(tileCoordinates[0] + 1));
             setButtonsHoverOverEvents(tile);
         }
     }
 
     private void createNewBoardTileAndAddIt(int absoluteXCoordinate, int absoluteYCoordinate) {
-        int boardIndex = AdjacentFieldsCalc.translateCoordinatesToIndex(absoluteXCoordinate - 1, absoluteYCoordinate - 1);
+        int boardIndex = AdjacentTilesCalc.translateCoordinatesToIndex(absoluteXCoordinate - 1, absoluteYCoordinate - 1);
```
Comment (piotr921) [inline src/main/java/com/spanish_inquisition/battleship/client/board/GameBoardBuilder.java]: better name 👍

File: src/main/java/com/spanish_inquisition/battleship/common/AdjacentFieldsCalc.java [hunk 5]
```diff
@@ -1,25 +0,0 @@
-package com.spanish_inquisition.battleship.common;
-
-import com.spanish_inquisition.battleship.client.board.GameBoardBuilder;
-
-/**
- * @author Michal_Partacz
- */
```
Comment (piotr921) [inline src/main/java/com/spanish_inquisition/battleship/common/AdjacentFieldsCalc.java]: delete @mprtcz code 👍

File: src/main/java/com/spanish_inquisition/battleship/common/AdjacentTilesCalc.java [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+            case LEFT: return getBoardIndexOfTileToTheLeft(boardIndex);
+            case RIGHT: return getBoardIndexOfTileToTheRight(boardIndex);
+            case ABOVE: return getBoardIndexOfTileAbove(boardIndex);
+            case BELOW: return getBoardIndexOfTileBelow(boardIndex);
+            case UPPER_LEFT: return getBoardIndexOfUpperLeftTile(boardIndex);
+            case UPPER_RIGHT: return getBoardIndexOfUpperRightTile(boardIndex);
+            case LOWER_LEFT: return getBoardIndexOfLowerLeftTile(boardIndex);
+            case LOWER_RIGHT: return getBoardIndexOfLowerRightTile(boardIndex);
+            default: return -1;
+        }
```
Comment (piotr921) [inline src/main/java/com/spanish_inquisition/battleship/common/AdjacentTilesCalc.java]: All this methods should be in enum.

File: src/test/java/com/spanish_inquisition/battleship/common/AdjacentTilesCalcTest.java [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+                {44, 35}, //random tile
+                {2, -1}, //tile from first row
+                {95, 86}, //tile from last row
+                {10, 1}, //first tile in row
+                {89, -1}, //last tile in row
+        };
+    }
+
+    @DataProvider(name = "LowerRightTileProvider")
+    public Object[][] lowerRightTileProvider() {
```
Comment (piotr921) [inline src/test/java/com/spanish_inquisition/battleship/common/AdjacentTilesCalcTest.java]: Private data providers, name in data provider is not nesessary

**Pull request merged**